tests/unit/test_orderedWinEventLimiter.py

```diff
@@ -151,6 +177,161 @@ def test_hideNoOverridesShow_whenDetailsDontMatch(self):
 		]
 		self.assertEqual(expectedEvents, actualEvents)
 
+	def test_alwaysAllowedObjects_specialCaseEvents(self):
+		# We have events from two unique objects:
+		# Window, objectID, childID
+		allowedSource = (1, 1, 1)
+
+		limiter = OrderedWinEventLimiter(maxFocusItems=4)
+		for n in range(2000):  # send many events, to saturate all limits.
+			eventId = specialCaseEvents[n % len(specialCaseEvents)]
+			limiter.addEvent(eventId, *allowedSource, threadID=0)
+		events = limiter.flushEvents(alwaysAllowedObjects=[allowedSource, ])
+
+		expected = [
+			# Two Foreground events, because they are added to multiple queues.
+			# Added to both _focusEventCache and _genericEventCache
+			# See also test_limitEventsPerThread
+			(winUser.EVENT_SYSTEM_FOREGROUND, *allowedSource),
+			(winUser.EVENT_SYSTEM_FOREGROUND, *allowedSource),

```

Comment:
I think it would be good to create a new "good first issue" to address this. The tests mean it is fairly well defined, the intent is easy to understand, and is unlikely to require a big change.